### PR TITLE
Clean up configure.ac

### PR DESCRIPTION
**Pull request by jubalh**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -183,32 +195,12 @@ CFLAGS="$CFLAGS_RESTORE"
...
-               [PKG_CHECK_MODULES([GTK], [gtk+-2.0 >= 2.24.10],
-                                  [AC_DEFINE([HAVE_GTK], [1], [libgtk module])],
-                                  [AC_MSG_ERROR([gtk+-2.0 >= 2.24.10 or gtk+-3.0 >= 3.24.0 is required for icons and clipboard])],
-                                  [AC_MSG_NOTICE([gtk+-3.0/gtk+2.0 not found, icons and clipboard not enabled])])])])])
+## Check for ncursesw/ncurses.h first, Arch linux uses ncurses.h for ncursesw
+AC_CHECK_HEADERS([ncursesw/ncurses.h], [], [])
+AC_CHECK_HEADERS([ncurses.h], [], [])
+AC_CHECK_HEADERS([curses.h], [], [])
 
+# Check for readline
```
Comment (jubalh) [inline configure.ac:203]: Can we replace this with `AX_LIB_READLINE` https://www.gnu.org/software/autoconf-archive/ax_lib_readline.html ? Seems like we are doing more in OSX case.

File: configure.ac [hunk 2]
```diff
@@ -258,6 +267,8 @@ AS_IF([test "x$PLATFORM" = xosx],
                     [AC_MSG_ERROR([libnotify is required but does not exist])],
                     [AC_MSG_NOTICE([libnotify support will be disabled])])])])])
 
+dnl feature: xscreensaver
+
 # TODO: rewrite this
```
Comment (jubalh) [inline configure.ac:272]: Rewrite this how? Why?

**Comment by jubalh** [comment]
@carlocab do you see why osx fails?
https://formulae.brew.sh/formula/autoconf looks like it has autoconf 2.71. We bump it to 2.69 in this PR.
Do we need to specify a version in the [brewfile](https://github.com/profanity-im/profanity/blob/master/Brewfile#L1) ?

**Comment by carlocab** [comment]
Is this your error?
```
src/ui/inputwin.c:150:21: error: incompatible function pointer types assigning to 'Function *' (aka 'int (*)(const char *, int)') from 'int (void)' [-Werror,-Wincompatible-function-pointer-types]
    rl_startup_hook = _inp_rl_startup_hook;
                    ^ ~~~~~~~~~~~~~~~~~~~~
src/ui/inputwin.c:408:5: error: implicit declaration of function 'rl_add_funmap_entry' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    rl_add_funmap_entry("prof_win_1", _inp_rl_win_1_handler);
    ^
src/ui/inputwin.c:444:5: error: implicit declaration of function 'rl_bind_keyseq' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    rl_bind_keyseq("\\e1", _inp_rl_win_1_handler);
    ^
src/ui/inputwin.c:444:5: note: did you mean 'rl_bind_key'?
```
It looks like a missing `#include` to me, and not from a different version of `autoconf`. You can try using Autoconf 2.69, though.

To do that, you can change this

https://github.com/profanity-im/profanity/blob/7bbaa11a54b6b32055df604f8bf5dd00fe97388d/Brewfile#L1

to
```ruby
brew 'autoconf@2.69'
```
Note that since this is an alternate version, it's not symlinked into `/usr/local/bin`. You need to do something like
```ruby
brew 'autoconf@2.69', link: true
```
and/or `brew unlink autoconf; brew link --force autoconf@2.69` right after doing `brew bundle`.

**Comment by jubalh** [comment]
> Is this your error?

Seems like it.

> It looks like a missing #include to me, and not from a different version of autoconf.

However master builds fine. I didn't change any includes.
Just trying to clean up the `configure.ac` in this PR. And I don't see what could cause this actually.

Sad that GH CI doesn't seem to build for each of the commits I did so I could see which one caused this.

**Comment by carlocab** [comment]
You could try passing `-Werror=incompatible-function-pointer-types -Werror=implicit-function-declaration` to see if you can reproduce this locally.

**Comment by jubalh** [comment]
Seems like OSX needs the readline as CPPFLAGS.

File: configure.ac (hunk 2 above)
Comment (jubalh) [reply configure.ac:272]: Will use PKG_CHECK_MODULES

**Review by sjaeckel — APPROVED** [review]

**Pull request merged**